caluma/caluma_core/tests/test_jexl.py

```diff
@@ -103,3 +103,81 @@ def test_jexl_cache():
 
     # validate invariants
     assert cache._cache.keys() == cache._mru.keys()
+
+
+@pytest.mark.parametrize(
+    "expression,result",
+    [
+        ("[{ key: 1 }]|mapby('key')", [1]),
+        ("[{ otherkey: 1 }]|mapby('key')", [None]),
+        ("[]|mapby('key')", []),
+    ],
+)
+def test_mapby_operator(expression, result):
+    assert JEXL().evaluate(expression) == result
+
+
+@pytest.mark.parametrize(
+    "expression,result",
+    [
+        ("[1,2] intersects [2,3]", True),
+        ("[1,2] intersects [3,4]", False),
+        ("[] intersects []", False),
+        ("[1] intersects []", False),
+        ("['foo'] intersects ['bar', 'bazz']", False),
+        ("['foo'] intersects ['foo', 'foo']", True),
+        ("[1] intersects [1] && [2] intersects [2]", True),
+        ("[2] intersects [1] + [2]", True),
+    ],
+)
+def test_intersects_operator(expression, result):
+    assert JEXL().evaluate(expression) == result
+
+
+@pytest.mark.parametrize(
+    "expression,result",
+    [
+        ("[1, 2, 3]|min", 1),
+        ("['a', 'b']|min", "a"),
+        ("[1, 2, '3']|min", None),
+        ("['a', None]|min", None),
+        ("[1, 2, 3]|max", 3),
+        ("['a', 'b']|max", "b"),
+        ("[1, 2, '3']|max", None),
```

Comment:
This for example should return `2` not `None`

Reply:
TBH I'm not entirely sure where to go with this regarding user friendlyness. Not saying following the JS implementation is wrong, but:

* Should we accept non-numbers at all?
* Should we try to convert non-number arguments to numbers first?
* What if we can't convert - fail or just ignore said argument?

I think we'd be in a better place if we maybe even used separate operators for string versus number variants of the functions: We could use `|asort` to sort alphabetically, followed by `|first` or `|last` to do the same for textual `min`/`max`. Then we could do some more sane checking/enforcing/converting for the numerical variants

Reply:
We already discussed this in detail...

1. Yes, we should. Imagine this: a calculation over a table field that is optional. Some values are bound to be `None` - this implementation would fail.
2. No, we should not. If someone tries to calculate with a text field and it suddenly works because we convert it it would confuse. Also, we'd need to do that for everything in those calculated fields to be "fair"

Reply:
@winged I had the same thoughts. But the application here is very domain specific so it makes sense to narrow it down.
@anehx python `sort` also sorts string alphabetically, that's why. But I don't really see the application in our jexls either.

Reply:
OK, I wasn't part of that discussion, so didn't know there's an explicit decision already. I only mentioned the string variants because we have test parametrisations for that just a few lines further up. Feel free to disregard my feedback